app/services/db/monthly_sp_auth_count/total_monthly_auth_counts_within_iaa_window.rb

```diff
@@ -19,32 +19,31 @@ def call(issuer:, iaa:, iaa_start_date:, iaa_end_date:)
 
         iaa_range = (iaa_start_date..iaa_end_date)
 
-        full_months, partial_months = Reports::MonthHelper.months(iaa_range).
-          partition do |month_range|
-            Reports::MonthHelper.full_month?(month_range)
-          end
-
-        # The subqueries create a uniform representation of data:
-        # - full months from monthly_sp_auth_counts
-        # - partial months by aggregating sp_return_logs
+        # Query a month at a time, to keep query time/result size fairly reasonable
         # The results are rows with [user_id, ial, auth_count, year_month]
-        queries = [
-          *full_month_subquery(issuer: issuer, full_months: full_months),
-          *partial_month_subqueries(issuer: issuer, partial_months: partial_months),
-        ]
+        months = Reports::MonthHelper.months(iaa_range)
+        queries = build_queries(issuer: issuer, months: months)
 
         ial_to_year_month_to_users = Hash.new do |ial_h, ial_k|
           ial_h[ial_k] = Hash.new { |ym_h, ym_k| ym_h[ym_k] = Multiset.new }
         end
 
         queries.each do |query|
-          stream_query(query) do |row|
-            user_id = row['user_id']
-            year_month = row['year_month']
-            auth_count = row['auth_count']
-            ial = row['ial']
-
-            ial_to_year_month_to_users[ial][year_month].add(user_id, auth_count)
+          temp_copy = ial_to_year_month_to_users.deep_dup
+
+          with_retries(
+            max_tries: 3,
+            rescue: PG::TRSerializationFailure,
+            handler: proc { ial_to_year_month_to_users = temp_copy },
+          ) do
```

Comment:
see commit notes (70ae9c6f370152adf6306c57d53bb160983cfafd) for a longer comment, but short version is we get these errors occasionally and I figured it was worth a shot seeing if we could quickly retry + recover rather than abort the entire job

Reply:
and in case it's not obvious what it's doing, we have a ruby nested hash/multiset thing object where we add results incrementally and this is creating a copy before streaming the results, and then restoring the copy every time the query fails 

we could rewrite this as `begin/rescue` syntax but we'd need add some sleep code and need a retry counter ourself, so this seemed clear enough? the alternative would be:


```ruby
temp_copy = ial_to_year_month_to_users.deep_dup
attempt_count = 0
begin
  stream_query(query) do |row|
    # ...
  end
rescue PG::TRSerializationFailure => e
  attempt_count += 1
  if attempt_count <3
    ial_to_year_month_to_users = temp_copy
    retry
   else
    raise e
  end
end
```